src/Bicep.Core/Syntax/FunctionCallSyntaxBase.cs

```diff
@@ -16,7 +16,7 @@ protected FunctionCallSyntaxBase(IdentifierSyntax name, Token openParen, IEnumer
             this.OpenParen = openParen;
             this.Children = children.ToImmutableArray();
             this.CloseParen = closeParen;
-            this.Arguments = children.OfType<FunctionArgumentSyntax>().ToImmutableArray();
```

Comment:
This is a hidden bug - calling `ToImmutableArray` on an `IEnumerable<T>` twice can be problematic if the source is a generator method, which happens to be the case when the syntax is created by the decompiler. The bug was identified by some new test cases of this PR.

Reply:
Is this the right fix?  Should something instead be changed in the decompiler?  Kinda feels like this breaks an implicit understanding of how enumerables work.

Reply:
... my concern being that this will pop up again sometime later in similar scenarios.

Reply:
>  Kinda feels like this breaks an implicit understanding of how enumerables work.

Could you share what the implicit understanding is?

I think the fix is right for any method that accepts IEnumerable. Multiple enumerations of an IEnumerable collection is generally considered as [a practice to be avoided](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1851) due to performance issues and unexpected behavior.

Reply:
Speaking of which, it might be worth enabling the code quality rules in our repo...

Reply:
Created https://github.com/Azure/bicep/issues/13521 to track it. Let's discuss it with the team later today.